### PR TITLE
XSVS plotting tools

### PR DESCRIPTION
**Pull request by sameera2004**

### CODE REVIEW

**Comment by sameera2004** [comment]
http://nbviewer.ipython.org/github/sameera2004/my_ipy_notebook/blob/master/matplotlib/Plotting_tools_XSVS.ipynb

File: xray_vision/xsvs_plotting.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+def mean_intensity_plotter(ax, mean_intensity_sets, index_list,
+                           title="Mean intensities",
+                           xlabel="Frames", ylabel="Mean Intensity"):
+    """
+    This will plot mean intensities for ROIS' of the labeled array
+    for different image sets.
+
+    Parameters
+    ----------
+     ax : Axes
```
Comment (tacaswell) [inline xray_vision/xsvs_plotting.py]: This is documented as a single Axes, but you are using this is a list below.

File: xray_vision/xsvs_plotting.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            total = 0
+            first = 0
+            x = [len(elm) for elm in mean_intensity_sets]
+            x_val = np.arange(sum(x))
+            for j in range(len(mean_intensity_sets)):
+                total += x[j]
+                ax[i].plot(x_val[first:total], mean_intensity_sets[j][:, i],
+                           label=str(j+1)+" image_set")
+                first = total
+        plt.legend()
```
Comment (tacaswell) [inline xray_vision/xsvs_plotting.py]: ax.legend()

File: xray_vision/xsvs_plotting.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        for i in range(len(index_list[0])):
+            ax[i].set_ylabel(ylabel)
+            ax[i].set_title(title+" for ROI " + str(i+1))
+            total = 0
+            first = 0
+            x = [len(elm) for elm in mean_intensity_sets]
+            x_val = np.arange(sum(x))
+            for j in range(len(mean_intensity_sets)):
+                total += x[j]
+                ax[i].plot(x_val[first:total], mean_intensity_sets[j][:, i],
```
Comment (tacaswell) [inline xray_vision/xsvs_plotting.py]: you should capture and return the artists.

File: xray_vision/xsvs_plotting.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        title of the plot
+
+    x_label : str, optional
+        x axis label
+
+    y_label : str, optional
+        y axis label
+
+    """
+    if np.all(map(lambda x: x == index_list[0], index_list)):
```
Comment (tacaswell) [inline xray_vision/xsvs_plotting.py]: I would do this as 

```python
if not cond:
    raise ...

other_code_here
```

File: xray_vision/xsvs_plotting.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    y_label : str, optional
+        y axis label
+    """
+    num_rois = combine_intensity.shape[1]
+    for i in range(num_rois):
+        ax.set_ylabel(ylabel)
+        ax.set_xlabel(xlabel)
+        ax.set_title(title)
+        ax.plot(combine_intensity[:, i], label=str(i+1)+" ROI")
+    plt.legend()
```
Comment (tacaswell) [inline xray_vision/xsvs_plotting.py]: `ax.legend()`  Please never use plt in a script, only at the repl

**Comment by tacaswell** [comment]
Please return the artists added in each plotting function (don't worry about the axis labels though).

**Comment by ericdill** [comment]
Can you move this into `xray_vision/mpl_plotting/speckle.py`?

**Comment by ericdill** [comment]
nevermind, I will take care of it

**Comment by ericdill** [comment]
see #68

**Pull request closed without merging**